cloudinit/sources/DataSourceAzure.py

```diff
@@ -529,6 +543,9 @@ def _get_data(self):
         except Exception as e:
             LOG.warning("Failed to get system information: %s", e)
 
+        if self.distro is not None:
```

Comment:
This if check is not needed; nothing is going to work if distro passed to a datasource is None.